volume/drivers/drivers.go

```diff
@@ -0,0 +1,143 @@
+package volumedrivers
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/pkg/plugins"
+)
+
+// Opts are options that are currently passed as a string map
+type Opts map[string]string
+
+// Driver is the Volume Driver interface for adapters that creates and removes Volumes
+type Driver interface {
+	// Create a new driver adapter
+	New(string, interface{}) Driver
```

Comment:
I believe this is following the design pattern of driver/adapter.  The return is a Driver instance, but this represents a driver that has been initialized and becomes an adapter at that point.